Add netcore50 configuration for System.IO

mellinoe opened the pull request:
As part of this, I've updated the buildtools version from 141 to 142. I needed to disable the inclusion of AssemblyInfoPartial as this library does not reference ResourceManager.

Also consolidated the source projects into a single file (deleted facade folder).

I did a quick pass of the source code to clean it and bring it in line with our guidelines. There are some additional cleanup items that we could tackle, like replacing some Contracts calls with Debug calls.

@weshaggard

weshaggard commented on src/System.IO/src/System.IO.builds:
```diff
@@ -3,9 +3,12 @@
       <TargetGroup>net46</TargetGroup>
     </Project>
+    <Project Include="System.IO.csproj">
+      <TargetGroup>netcore50aot</TargetGroup>
```
Please add `<OSGroup>Windows_NT</OSGroup>`

weshaggard commented on src/System.IO/src/System.IO.csproj:
```diff
@@ -6,18 +6,68 @@
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50aot_Release|AnyCPU'" />
+
+  <PropertyGroup Condition="'$(TargetGroup)' == 'net46'">
+    <ResourcesSourceOutputDirectory>None</ResourcesSourceOutputDirectory>
```
Should we add a proper way to disable this in BuildTools targets.

weshaggard commented on src/System.IO/src/System.IO.nuget.targets:
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Lets not commit this file.

mellinoe replied:
Agh, yes, this one always tries to sneak its way in. Will remove.

weshaggard commented on src/System.IO/src/System/IO/BinaryReader.cs:
```diff
@@ -0,0 +1,638 @@
+        public BinaryReader(Stream input, Encoding encoding, bool leaveOpen)
+        {
+            if (input == null)
+            {
```
Can you make the bracing style consistent? for the argument checks?

weshaggard commented on src/System.IO/src/System/IO/BinaryReader.cs:
```diff
@@ -0,0 +1,638 @@
+            if (minBufferSize < 16)
+                minBufferSize = 16;
+            _buffer = new byte[minBufferSize];
+            // m_charBuffer and m_charBytes will be left null.
```
Looks like a missed rename for m_charBuffer

weshaggard commented on src/System.IO/src/System/IO/BinaryReader.cs:
```diff
@@ -0,0 +1,638 @@
+            return ((ulong)hi) << 32 | lo;
+        }
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
```
You can remove the Security attributes.

weshaggard commented on src/System.IO/src/System/IO/Stream.cs:
```diff
@@ -0,0 +1,453 @@
+            return WriteAsync(buffer, offset, count, CancellationToken.None);
+        }
+
+        [ComVisible(false)]
```
You can remove the ComVisible attributes.

mellinoe commented:
I went through and normalized all of the braces in the new code that I moved out, and replaced the usages of Contract.Assert and Requires with Debug.Assert. I've run the System.IO tests for .NET Native and they all pass.

weshaggard commented:
LGTM
